libs/patchwheel/parse.go

```diff
@@ -0,0 +1,58 @@
+package patchwheel
+
+import (
+	"fmt"
+	"strings"
+	"time"
+)
+
+// WheelInfo contains information extracted from a wheel filename
+type WheelInfo struct {
+	Distribution string   // Package distribution name
+	Version      string   // Package version
+	Tags         []string // Python tags (python_tag, abi_tag, platform_tag)
+}
+
+// CalculateNewVersion generates a new version string and filename based on the wheel info and modification time.
+// The version is updated according to the following rules:
+//   - if there is an existing part after + it is dropped
+//   - append +<mtime of the original wheel> to version
+func CalculateNewVersion(info WheelInfo, mtime time.Time) (newVersion, newFilename string) {
```

Comment:
should this be a private function? Don't think it should be used outside of this package anyway

Reply:
sure, updated.